src/assets/Generator.Shared/NextLinkOperationImplementation.cs

```diff
@@ -215,34 +197,63 @@ internal static bool TryGetApiVersion(Uri startRequestUri, out ReadOnlySpan<char
         /// <summary>
         /// This function is used to get the final request uri after the lro has completed.
         /// </summary>
-        private string? GetFinalUri()
+        private string? GetFinalUri(Response response)
         {
-            // Set final uri as null if the response header doesn't contain "Operation-Location" or "Azure-AsyncOperation".
+            // Set final uri as null if the response for initial request doesn't contain header "Operation-Location" or "Azure-AsyncOperation".
             if (_headerSource is not (HeaderSource.OperationLocation or HeaderSource.AzureAsyncOperation))
```

Comment:
Here we instantly return `null` if initial request doesn't contain `Operation-Location` or `Azure-Async-Operation`. Then why do we check `OperationFinalStateVia.Location` and `OperationFinalStateVia.OriginalUri` below?